morgoth/addons/tests/test_api.py

```diff
@@ -128,6 +128,16 @@ def test_it_can_add_addons_to_group(self, api_client):
         assert res.status_code == 204
         assert group.addons.count() == 3
 
+    def test_adding_an_addon_that_doesnt_exist(self, api_client):
+        group = AddonGroupFactory()
+
+        res = api_client.post('/api/v1/addon_group/%s/add_addons/' % group.id, {
+            'addon_ids': [1]
+        })
+
+        assert res.status_code == 204
```

Comment:
I think this should be a 404 error. Requests that return 20x but don't do what you asked would confuse me.